tests/test_authz_decorator.py

```diff
@@ -1,167 +1,81 @@
+    @staticmethod
+    def get_guest_authorization() -> dict:
+        return {"allow": {"guest_res": {"handler": {"allow": "*"}}}, "deny": {}}
 
-            def diff_perm(self):
-                pass
+    @add_route("/")
+    @authorization()
+    def handler(self, restrictions) -> Response:  # pylint: disable=unused-argument
+        return Response("ok")
 
-        auth_header = Authorizer.sign_authz(
-            {"allow": {"test_res": {"handler": {"allow": "*"}}}, "deny": {}},
-            SAMPLE_PRIVATE_KEY,
-        )
-        res = XResource({**event, "headers": {"authorization": auth_header}})
+    @add_route("/garbage2")
+    @authorization()
+    def garbage(self, restrictions) -> Response:  # pylint: disable=unused-argument
+        return Response("ok")
 
-        assert check_permission(res, "handler") == {"allow": "*", "deny": None}
-        with pytest.raises(PermissionDenied):
-            check_permission(res, "garbage")
 
-        res = XResource({**event})
-        with pytest.raises(Unauthorized):
-            check_permission(res, "diff_perm")
+class TestAuthorizationDecoratorGuestPermissions:
+    def test_get_success(self) -> None:
+        res_instance = GuestResource(Event("/", "GET"))
+        assert res_instance().status_code == HTTPStatus.OK
 
-    def test_check_permission_guest_policy(self) -> None:
-        class XResource(Resource):
-            _name = "test_res"
+    def test_limited_permissions_failed(self) -> None:
+        res_instance = GuestResource(Event("/garbage2", "GET"))
+        assert res_instance().status_code == HTTPStatus.FORBIDDEN
 
-            @add_route("/")
-            @authorization()
-            def handler(self, restrictions):
-                assert restrictions == {"allow": "*", "deny": None}
-                return Response("x")
-
-            def get_guest_authorization(self) -> dict:
-                return {"allow": {"*": "*"}, "deny": {}}
-
-        res = XResource({**event})
-        assert check_permission(res, "handler") == {"allow": "*", "deny": None}
-
-    @patch.dict(environ, env_mock)
-    def test_authorization_header_takes_priority_over_guest_policy_when_both_present(self, *_args):
-        class XResource(Resource):
-            @add_route("/")
-            @authorization()
-            def handler(self, restrictions):
-                assert restrictions == {"allow": "*", "deny": None}
-                return Response("x")
-
-            @staticmethod
-            def get_guest_authorization() -> dict:
-                return {"allow": {}, "deny": {}}
-
-        auth_header = Authorizer.sign_authz({"allow": "*", "deny": {}}, SAMPLE_PRIVATE_KEY)
-        resp = XResource({**event, "headers": {"authorization": auth_header}})()
-        assert resp.status_code == 200
+    def test_inharitance_success(self, full_access_auth_header) -> None:
+        res_instance = GuestResource(
+            Event("/garbage2", "GET", headers={"authorization": full_access_auth_header})
```

Comment:
This test does not confirm inheritance 😕  You should pass the Authorization header which does not have access to the `handler` method and thanks to inheritance it will get it anyway because it is declared in the guest authorization:
```python
def test_inheritance_success(self, limited_access_auth_header: str) -> None:
    # the limited access auth header grants access only to the `test_res`
    event = Event("/", "GET", headers={"authorization": limited_access_auth_header})
    res_instance = GuestResource(event)

    # no matter the user accesses, they will get the OK response because of guest permissions
    assert res_instance().status_code == HTTPStatus.OK
```